.github/workflows/ci.yml

```diff
@@ -36,3 +36,10 @@ jobs:
     - run: cargo test --verbose --all --all-features
     - run: cargo test --verbose --all
 
+    # Test using minimal dependency versions from Cargo.toml
+    - if: matrix.rust == 'nightly'
+      run: |
+        cargo +nightly update -Z minimal-versions
+        cargo check --all-features --verbose --all-targets
```

Comment:
Given that this is basically just a Will It Compile test, I suppose adding the `--tests` (and possibly `--examples`) options to this `cargo check` and dropping the `cargo test` below will be sufficient and save some cycles.

Reply:
The separate `check` is indeed not super useful, though separate `build` and `test` is convenient to quickly spot where things go wrong. (Note that this was inspired by the ci.yml in the domain crate, so we might want to have a look at that one as well).
On saving cycles, it might actually make more sense to get rid of the beta/nightly runs.